examples/binaries/constructor/src/builder.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{Arg, Call};
+use alloc::{boxed::Box, string::ToString, vec::Vec};
+use core::{fmt::Debug, ops::Deref};
+use parity_scale_codec::{WrapperTypeDecode, WrapperTypeEncode};
+
+#[derive(Default, Debug, Clone)]
+pub struct Calls(Vec<Call>);
+
+impl From<Vec<Call>> for Calls {
+    fn from(calls: Vec<Call>) -> Self {
+        Self(calls)
+    }
+}
+
+impl Deref for Calls {
+    type Target = Vec<Call>;
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_ref()
+    }
+}
+
+impl WrapperTypeEncode for Calls {}
+impl WrapperTypeDecode for Calls {
+    type Wrapped = Vec<Call>;
+}
+
+impl Calls {
+    pub fn builder() -> Self {
+        Default::default()
+    }
+
+    pub(crate) fn calls(self) -> Vec<Call> {
+        self.0
+    }
+
+    pub fn push(mut self, call: Call) -> Self {
+        self.0.push(call);
+        self
+    }
+
+    pub fn push_many<const N: usize>(mut self, calls: [Call; N]) -> Self {
```

Comment:
```suggestion
    pub fn push_many<const N: usize>(mut self, calls: impl Iterator<Item = Call>) -> Self {
```

Reply:
kept both
for me seems more neat using just arrays [val, val], instead of [val, val].iter()

Reply:
iterator is a more general approach, very likely to be in such api